From 3509c2ad8cd625a3d6306401da036d7bebf47b42 Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Wed, 06 Jul 2022 17:27:24 +0300
Subject: [PATCH] [SW-94367] gaudi2: add srns mode configuration

This patch adds SRNS mode configuration as part of pcie link up
routine. SRNS mode configuration is performed based on dump_sel[0]
value, refclk mode and pll short lock enable registers are cleared
as recommended by arch team.

Change-Id: I752a51b6d5452472b2d7b8ff3cc033db606a862a
---

diff --git a/arch/gaudi3/arc/common.c b/arch/gaudi3/arc/common.c
index 36fd052..0a51b3d 100644
--- a/arch/gaudi3/arc/common.c
+++ b/arch/gaudi3/arc/common.c
@@ -288,9 +288,17 @@
 	return !!(dump_sel & DUMP_SEL_SSC_MASK);
 }
 
+#define DUMP_SEL_SRNS_MASK	0x1
 uint8_t get_srns_mode_from_strap(void)
 {
-	return 0;
+	uint32_t boot_strap = regmap_readl(mmD0_PSOC_BOOT_CONF_BOOT_STRAP_PINS_L);
+
+	uint32_t dump_sel = (boot_strap &
+		D0_PSOC_BOOT_CONF_BOOT_STRAP_PINS_L_DUMP_SEL_M) >>
+		D0_PSOC_BOOT_CONF_BOOT_STRAP_PINS_L_DUMP_SEL_S;
+
+	/* dumpsel[0] is SRNS */
+	return !!(dump_sel & DUMP_SEL_SRNS_MASK);
 }
 
 void pcie_phy_ssc_en_set(void)
diff --git a/common/pcie.c b/common/pcie.c
index 0ed6d4c..6e51efe 100644
--- a/common/pcie.c
+++ b/common/pcie.c
@@ -103,6 +103,11 @@
 			regmap_writel(1, mmAPP_SRIS_MODE);
 		}
 
+		if (get_srns_mode_from_strap()) {
+			pcie_phy_set_lane_refclk_mode();
+			pcie_phy_set_pll_short_lock();
+		}
+
 		pcie_wa_snps_case_01038891();
 		pcie_phy_ssc_en_set();
 
